File: sol/solution/0001-0100/0005. Longest Palindromic Substring/Solution.rs
impl Solution {
    pub fn longest_palindrome(s: String) -> String {
        let n = s.len();
        let chars: Vec<char> = s.chars().collect();
        let mut start = 0;
        let mut max_len = 1;   
        for i in 0..n {
            let mut left = i;
            let mut right = i;
            while left < n && right < n && chars[left] == chars[right] {
                if right - left + 1 > max_len {
                    start = left;
                    max_len = right - left + 1;
                }
                if left == 0 { break; }
                left -= 1;
                right += 1;
            }
            left = i;
            right = i + 1;
            while left < n && right < n && chars[left] == chars[right] {
                if right - left + 1 > max_len {
                    start = left;
                    max_len = right - left + 1;
                }
                if left == 0 { break; }
                left -= 1;
                right += 1;
            }
        }
        s[start..start + max_len].to_string()
    }
}

File: sol/solution/0001-0100/0005. Longest Palindromic Substring/Solution.go
func longestPalindrome(s string) string {
    n := len(s)
    start, maxLen := 0, 1
    for i := 0; i < n; i++ {
        left, right := i, i
        for left >= 0 && right < n && s[left] == s[right] {
            if right - left + 1 > maxLen {
                start = left
                maxLen = right - left + 1
            }
            left--
            right++
        }
        left, right = i, i+1
        for left >= 0 && right < n && s[left] == s[right] {
            if right - left + 1 > maxLen {
                start = left
                maxLen = right - left + 1
            }
            left--
            right++
        }
    }
    return s[start : start+maxLen]
}